src/audiomixerboard.cpp

```diff
@@ -1210,92 +1210,98 @@ void CAudioMixerBoard::ApplyNewConClientList ( CVector<CChannelInfo>& vecChanInf
 
         // search for channels with are already present and preserve their gain
         // setting, for all other channels reset gain
-        for ( int i = 0; i < MAX_NUM_CHANNELS; i++ )
+
+        // get all channels which are in use/not in use. We use the array index of vecChanInfo
+        // if the fader is in use, else INVALID_INDEX to specify it is not in use
+        int iFaderNumber[MAX_NUM_CHANNELS];
+
+        for ( int iChanID = 0; iChanID < MAX_NUM_CHANNELS; iChanID++ )
+        {
+            iFaderNumber[iChanID] = INVALID_INDEX;
+        }
+
+        for ( int iFader = 0; iFader < iNumConnectedClients; iFader++ )
         {
-            bool bFaderIsUsed = false;
+            iFaderNumber[vecChanInfo[iFader].iChanID] = iFader;
+        }
+
+        for ( int iChanID = 0; iChanID < MAX_NUM_CHANNELS; iChanID++ )
+        {
+            if ( iFaderNumber[iChanID] == INVALID_INDEX )
+            {
+                // fader is not used --> hide it
+                // before hiding the fader, store its level (if some conditions are fulfilled)
+                StoreFaderSettings ( vecpChanFader[iChanID] );
 
-            for ( int j = 0; j < iNumConnectedClients; j++ )
+                vecpChanFader[iChanID]->Hide();
+                continue;
+            }
+
+            // current fader is used
+            // check if fader was already in use -> preserve gain value
```

Comment:
This comment (line 1241) should move inside the `if`.  Then the "check if" can be removed.  Also ... "-> preserve gain value" appears to refer to the code at line 1277 - so really shouldn't be here: there's too much else happening first.